server/src/test/java/org/elasticsearch/index/translog/TestTranslog.java

```diff
@@ -53,6 +55,31 @@
 public class TestTranslog {
     static final Pattern TRANSLOG_FILE_PATTERN = Pattern.compile("translog-(\\d+)\\.tlog");
 
+    /*TODO currently there are two methods for corrupting translog.
+    There are two differences:
+     1) corruptTranslogFiles method is expecting other index files, not only translog.
+     2) corruptTranslogFiles method sometimes does not corrupt translog at all.
+    //It would be nice to have single corruption method to be used in integration and unit tests*/
+    /**
+     * Randomly overwrite some bytes in the translog files
+     */
+    public static void corruptTranslogFilesReliably(Logger logger, Random random, Path directory) throws Exception {
```

Comment:
Can we get rid of this and just use the other implementation, which looks more reliable than this one. If it is not more reliable, can you look into why?